evalml/tests/component_tests/test_components.py

```diff
@@ -534,6 +536,8 @@ def test_transformer_transform_output_type(X_y_binary):
 
             component.fit(X, y=y)
             transform_output = component.transform(X, y=y)
+            if 'sampler' in component.name:
+                transform_output = transform_output[0]
```

Comment:
Undersampler outputs both X and y in transform, unlike other transformers that output just X

Reply:
Agree, I think it would be nice to have the `transformer.transform()` method return the same number of outputs in all cases.

Reply:
Thank you for handling this while I was sweating and delusional!  Sounds good.